prime-router/metadata/hl7_mapping/common/observation-result.yml

```
@@ -165,30 +165,30 @@ elements:
     hl7Spec: [ '%{hl7OBXField}-19' ]
 
   - name: performing-organization-name-pracrole
-    condition: '%observation.performer.resolve() is PractitionerRole'
-    resource: '%resource.performer.resolve().organization.resolve()'
+    condition: '%observation.performer.resolve().ofType(PractitionerRole).exists() and %observation.performer.resolve().ofType(PractitionerRole).organization.exists()'
+    resource: '%resource.performer.resolve().ofType(PractitionerRole).organization.resolve()'
     schema: ./datatype/xon-organization
     constants:
       hl7OrgField: '%{hl7OBXField}-23'
 
   - name: performing-organization-name-non-pracrole
-    condition: '%observation.performer.resolve().is(PractitionerRole).not()'
-    resource: '%resource.performer.resolve()'
+    condition: '%observation.performer.resolve().ofType(PractitionerRole).exists().not() or %observation.performer.resolve().ofType(PractitionerRole).organization.exists().not()'
```

This again is fine and works, but it might be simpler to go with the following here and in `performing-organization-address-non-pracrole` (also optional):
```suggestion
    condition: '%observation.performer.resolve().ofType(Organization).exists()'
```

---

This is condition came down to which one should have priority, the current check will only use `Organization` if nothing is set via practionerRole